Model/Site.hs

```diff
@@ -42,3 +46,20 @@ destroySite siteId = do
 
 overSiteQuota :: [a] -> Plan -> Bool
 overSiteQuota xs Plan{..} = length xs >= planSiteQuota
+
+siteBranded :: SiteId -> DB Bool
+siteBranded siteId = do
```

Comment:
Given a site's ID, it seems like we should be able to answer the question in a single SQL query.